src/python/ensembl/io/genomio/integrity.py

```diff
@@ -21,211 +21,147 @@
 import hashlib
 import json
 from math import floor
-from os import path
+from os import PathLike
 from pathlib import Path
 import re
 import sys
 from typing import Any, Dict, List
 
 import argschema
 from BCBio import GFF
-from Bio import SeqIO
+from Bio import SeqIO, SeqFeature
 
 from ensembl.io.genomio.utils.archive_utils import open_gz_file
 from ensembl.io.genomio.utils.json_utils import get_json
 
 
+Lengths = Dict[str, int]
+
+
 class InvalidIntegrityError(Exception):
     """When a file integrity check fails"""
 
 
-class IntegrityTool:
-    """Check the integrity of sequence and annotation files in the genome"""
+class Manifest:
+    """Representation of the manifest and its files."""
+
+    def __init__(self, manifest_path: PathLike) -> None:
+        self.manifest_files = self.get_manifest(manifest_path)
+        self.genome: Dict[str, Any] = {}
+        self.seq_regions: Dict[str, Any] = {}
+
+        self.lengths: Dict[str, Dict] = {
```

Comment:
Following my previous comment, the name of this attribute can be mixed with the type defined above, and it seems to me they may be related, i.e. `Dict[str, Lengths]`?

Reply:
Genome and seq_regions do not contain lengths: they contain the actual data from their respective json files (hence why they are not in self.lengths).

Reply:
Yes that makes sense, thank you!